eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionSummaryWriter.java

```diff
@@ -0,0 +1,126 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.pipeline.io;
+
+import uk.ac.ebi.eva.accession.core.ISubmittedVariant;
+import uk.ac.ebi.eva.accession.core.SubmittedVariant;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class AccessionSummaryWriter {
+
+    private static final String VCF_MISSING_VALUE = ".";
+
+    private FastaSequenceReader fastaSequenceReader;
+
+    private boolean headerWritten;
+
+    private final FileWriter fileWriter;
+
+    private String accessionPrefix;
+
+    public AccessionSummaryWriter(File output, FastaSequenceReader fastaSequenceReader) throws IOException {
+        this.fastaSequenceReader = fastaSequenceReader;
+        this.headerWritten = false;
+        this.fileWriter = new FileWriter(output);
+        this.accessionPrefix = "ss";
+    }
+
+    public String getAccessionPrefix() {
+        return accessionPrefix;
+    }
+
+    public void setAccessionPrefix(String accessionPrefix) {
+        this.accessionPrefix = accessionPrefix;
+    }
+
+    public void write(Map<Long, ISubmittedVariant> accessions) throws IOException {
+        if (!headerWritten) {
+            writeHeader();
+            headerWritten = true;
+        }
+        for (Map.Entry<Long, ISubmittedVariant> variant : accessions.entrySet()) {
+            writeVariant(variant.getKey(), variant.getValue());
+        }
+        fileWriter.flush();
+    }
+
+    private void writeHeader() throws IOException {
+        fileWriter.write("##fileformat=VCFv4.1\n");
+        fileWriter.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n");
+    }
+
+    private void writeVariant(Long id, ISubmittedVariant normalizedVariant) throws IOException {
+        ISubmittedVariant variant = denormalizeVariant(normalizedVariant);
+        String vcfLine = variantToVcfLine(id, variant);
+        fileWriter.write(vcfLine);
+    }
+
+    private ISubmittedVariant denormalizeVariant(ISubmittedVariant normalizedVariant) {
+        if (normalizedVariant.getReferenceAllele().isEmpty() || normalizedVariant.getAlternateAllele().isEmpty()) {
+            if (fastaSequenceReader.doesContigExist(normalizedVariant.getContig())) {
+                return createVariantWithContextBase(normalizedVariant);
+            } else {
+                throw new IllegalArgumentException("Contig '" + normalizedVariant.getContig()
+                                                           + "' does not appear in the fasta file ");
```

Comment:
mmh, not sure. realistically, a warning will go unseen, and if the fasta is not complete, we need to know it to manage the accession DB and the EVA DB.

Furthermore, the very last writer should not the only place to check for this. The VCF should have been checked beforehand, so this will only fail if something was done wrong

Reply:
also, the accessioning system allows to accession the same variants again giving the same results, so it's not a big deal to fail fast.